packages/core/src/interpreter.ts

```diff
@@ -129,160 +106,132 @@ export class Interpreter<
 
   // Actor Ref
   public _parent?: ActorRef<any>;
-  public name: string;
   public ref: ActorRef<TEvent>;
+  private _actorContext: ActorContext<TEvent, SnapshotFrom<TBehavior>>;
 
   /**
    * The globally unique process ID for this invocation.
    */
   public sessionId: string;
-  private forwardTo: Set<string> = new Set();
+
+  // TODO: remove
+  public _forwardTo: Set<AnyActorRef> = new Set();
 
   /**
-   * Creates a new Interpreter instance (i.e., service) for the given machine with the provided options, if any.
+   * Creates a new Interpreter instance (i.e., service) for the given behavior with the provided options, if any.
    *
-   * @param machine The machine to be interpreted
+   * @param behavior The behavior to be interpreted
    * @param options Interpreter options
    */
-  constructor(
-    public machine: StateMachine<
-      TContext,
-      TEvent,
-      any,
-      any,
-      TResolvedTypesMeta
-    >,
-    options?: InterpreterOptions
-  ) {
+  constructor(public behavior: TBehavior, options?: InterpreterOptions) {
     const resolvedOptions = {
       ...defaultOptions,
       ...options
-    } as Required<InterpreterOptions>;
+    };
 
     const { clock, logger, parent, id } = resolvedOptions;
+    const self = this;
 
-    const resolvedId = id !== undefined ? id : machine.key;
-
-    this.name = this.id = resolvedId;
+    // TODO: this should come from a "system"
+    this.sessionId = registry.bookId();
+    this.id = id ?? this.sessionId;
     this.logger = logger;
     this.clock = clock;
     this._parent = parent;
-
     this.options = resolvedOptions;
-
     this.ref = this;
-
-    this.sessionId = this.ref.name;
-  }
-
-  public get initialized() {
-    return this.status === InterpreterStatus.Running;
-  }
-
-  public get initialState(): State<TContext, TEvent, TResolvedTypesMeta> {
-    const initialState =
-      this._initialState ||
-      ((this._initialState = this.machine.getInitialState()),
-      this._initialState);
-    return initialState;
+    this._actorContext = {
+      self,
+      id: this.id,
+      sessionId: this.sessionId,
+      logger: this.logger,
+      exec: (fn) => {
+        if (self.status === InterpreterStatus.NotStarted) {
+          // this._deferred.push(fn);
```

Comment:
Without this we fail this very important test (that we should add):
```ts
  it(`should execute entry actions when starting the actor after reading its snapshot first`, () => {
    const spy = jest.fn();
    
    const actorRef = interpret(
      createMachine({
        entry: spy
      })
    );

    actorRef.getSnapshot();

    actorRef.start();

    expect(spy).toHaveBeenCalled();
  });
```

Reply:
Test added 👍